wagtail/api/v2/serializers.py

```diff
@@ -40,7 +40,9 @@ def get_attribute(self, instance):
         return instance
 
     def to_representation(self, obj):
-        return type(obj)._meta.app_label + '.' + type(obj).__name__
+        name = type(obj)._meta.app_label + '.' + type(obj).__name__
+        self.context['view'].seen_types[name] = type(obj)
```

Comment:
I think serializers / fields shouldn't "know" about view from point of view of semantics. Serializers can (theoretically) be used on only with views (in workers for example).

But at the moment I don't see a better solution. I will think about it.